physical/raft/raft.go

```diff
@@ -359,6 +359,28 @@ type SetupOpts struct {
 	// StartAsLeader is used to specify this node should start as leader and
 	// bypass the leader election. This should be used with caution.
 	StartAsLeader bool
+
+	// RecoveryModeConfig is the configuration for the raft cluster in recovery
+	// mode.
+	RecoveryModeConfig *raft.Configuration
+}
+
+func (b *RaftBackend) StartRecoveryCluster(ctx context.Context, peer Peer) error {
+	recoveryModeConfig := &raft.Configuration{
+		Servers: []raft.Server{
+			{
+				ID: raft.ServerID(peer.ID),
+				// Use dummy address as there won't be any network interactions
+				// in recovery mode
+				Address: raft.ServerAddress(b.NodeID()),
```

Comment:
I think this should be the real cluster addr so when we switch out of recovery mode it will have the correct address.

Reply:
Likely it would need to be passed in via the peer object too.